### DISCUSSION TITLE
Best way to detect an idle state?

### DISCUSSION THREAD

**Discussion by jkbbwr** [tokio-rs/axum#2291] [Q&A] [answered]
I want to be able to detect if an axum service hasn't gotten any requests in the last (configurable) 5 minutes.

I was thinking of adding a middleware that ran before everything that basically sent a message to a task that loops forever reading messages or timing out, but I was wondering how expensive this might be or if its the right way to solve this.

Anyone got any ideas?

**Comment by davidpdrsn** [comment, marked as answer]
That sounds like a reasonable approach to me.

**Reply by jkbbwr** [reply]
Not too much overhead for it being on every request? 

Itd be sending a message down a channel to a task that is waiting for either recv on that channel to timeout, or get a message and loop.

**Reply by davidpdrsn** [reply]
Benchmark it.

**Reply by frenetisch-applaudierend** [reply]
If you're worried about sending too many messages, and you're willing to sacrifice some precision in your timeout, you could throttle the requests sent from your middleware. I.e. check if you have sent a request in the last 5 seconds or so already, and if so skip sending the message for this request.

**Reply by davidpdrsn** [reply]
I doubt it’s gonna matter much. You’re probably doing a bunch of IO (like talking to database) while handling the request. That’s gonna take way more time than sending on a channel.

**Reply by jkbbwr** [reply]
Ill stick tracing and metrics on it and keep an eye if its the most expensive part of the request. Thanks for the feedback
